lib/smart_proxy_remote_execution_ssh/runners/script_runner.rb

```diff
@@ -169,9 +169,12 @@ def prepare_start
     def initialization_script
       su_method = @user_method.instance_of?(SuUserMethod)
       # pipe the output to tee while capturing the exit code in a file
-      <<-SCRIPT.gsub(/^\s+\| /, '')
-      | sh -c "(#{@user_method.cli_command_prefix}#{su_method ? "'#{@remote_script} < /dev/null '" : "#{@remote_script} < /dev/null"}; echo \\$?>#{@exit_code_path}) | /usr/bin/tee #{@output_path}
-      | exit \\$(cat #{@exit_code_path})"
+      <<~SCRIPT
+        sh <<EOF | /usr/bin/tee #{@output_path}
+        #{@user_method.cli_command_prefix}#{su_method ? "'#{@remote_script} < /dev/null '" : "#{@remote_script} < /dev/null"}
+        echo \\$?>#{@exit_code_path}
+        EOF
+        exit $(cat #{@exit_code_path})
```

Comment:
This is interesting. It seems to break job cancellation in case there are spaces in remote_working_dir.

```
# Before cancel
$ pstree -p 383
ash(383)-+-sh(384)---busybox(386)---sleep(387)
         `-tee(385)

# After cancel
$ pstree -p 383
ash(383)---tee(385)

# After cancel, here we can see sleep(387) being a child of init(1)
$ pstree -p 1
sshd(1)-+-sleep(387)
        `-sshd(378)---ash(383)---tee(385)
```

The shell process gets killed, but whatever was running inside it gets re-parented under init, where it continues to run.

How about instead of killing by name, we would write the parent process ID to a special location and then on cancel we would read the pid and send the signal directly?